### DISCUSSION TITLE
Select render mode

### DISCUSSION THREAD

**Discussion by blippyblips** [google/model-viewer#2834] [Ideas]
Would like to add/have a way to force unlit even if it's not in the metadata of the gltf/glb models.
Maybe even the possibility to swap between them. Do you have some ideas on where you would start to do this in your code base?

**Comment by ghost** [comment]
Unlit VS Lit all depends on the how the material extension is processed in the [GLTFLoader](https://github.com/mrdoob/three.js/blob/master/examples/jsm/loaders/GLTFLoader.js), I don't believe we have any direct way to force this during loading but you could look at the loading process to gather some ideas.

I suppose if you wanted to force a material (after it's been loaded) to behave like it's unlit you would have to remove textures and set colors, the unlit extension actually provides some [guidance](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_materials_unlit/README.md#extension-compatibility-and-fallback-materials) on this:
> Implementation Note: For best fallback behavior in clients that do not implement the KHR_materials_unlit extension, authoring tools may use:
metallicFactor is 0 and emissiveFactor is [0, 0, 0].
roughnessFactor is greater than 0.5.
Omit metallicRoughnessTexture, occlusionTexture, emissiveTexture, and normalTexture.

I think it's a just a matter of using the existing material API to force those settings.
